File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> d;
        for (const auto n: nums) {
            d[n]++;
        }

        vector<pair<int, int>> count_pairs;
        for (const auto &pair: d) {
            count_pairs.push_back({pair.second, pair.first});
        }

        sort(count_pairs.rbegin(), count_pairs.rend());

        vector<int> arr;
        for (int i=0; i<k; i++) {
            arr.push_back(count_pairs[i].second);
        }
        return arr;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        a = [1] * n

        l = 1
        for i in range(n):
            a[i] *= l
            l *= nums[i]

        r = 1
        for i in range(n -1, -1, -1):
            a[i] *= r
            r *= nums[i]
        return a
        